py3status/modules/xscreensaver.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+"""
+Toggle and lock Xscreensaver.
+
+This is a partial rewrite of dpms.py. Most credit goes to the original author
+Andre Doser. This module shows and controls the status of Xscreensaver
+
+This script is useful for people who let Xscreensaver manage dpms settings.
+Xscreensaver has its own dpms variables separate from xset. Dpms can be
+safely turned off in xset as long as xscreensaver is running.
+Settings can be managed using "xscreensaver-demo".
+
+Configuration parameters:
+    cache_timeout: refresh interval for this module (default 60)
+    format: default format for this module (default '{output}')
+    string_off: display when Xscreensaver is not running (defaults '🖥️')
+    string_on: display when Xscreensaver is running (defaults '🖥️')
+
+ Format placeholders:
+    {output} display xscreensaver output
+
+@author neutronst4r <c7420{at}gmx{dot}net>
+"""
+from os import setpgrp
+from subprocess import Popen, DEVNULL, check_output, CalledProcessError
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    cache_timeout = 60
+    format = '{output}'
+    string_off = u'🖥️'
+    string_on = u'🖥️'
+
+    def xscreensaver(self):
+        """
+        """
+        self.run = self.py3.check_commands(['pidof xscreensaver'])
+        response = {
+            'cached_until': self.py3.time_in(self.cache_timeout),
+            'full_text': self.string_on,
+            'color': self.py3.COLOR_GOOD
+        }
+
+        try:
+            self.run = check_output(['pidof', 'xscreensaver'])
+        except CalledProcessError:
+            response = {
+                'cached_until': self.py3.time_in(self.cache_timeout),
+                'full_text': self.string_off,
+                'color': self.py3.COLOR_BAD
+            }
+
+        return response
+
+    def on_click(self, event):
+        """
+        """
+        if event['button'] == 1:
+            if self.run:
```

Comment:
this whole `self.run` stuff feels wrong.  we should just see if `xscreensaver` is running when we need to know - storing it just makes us more likely to have incorrect behaviour.